jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -65,29 +75,74 @@
      * When an entry can't be found in the cache, the global list is iterated
      * normally so the cache has no visible effect besides performance.
      */
```

Comment:
You should move this comment around since you also moved the ThreadLocal variable it speaks about.